nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/fingerprint/FingerprintFactory.java

```diff
@@ -565,14 +570,13 @@ private String getLoggableRepresentationOfSensitiveValue(String encryptedPropert
 
     private void initializeSensitivePropertyKeyBytes() {
         if (sensitivePropertyKeyBytes == null || sensitivePropertyKeyBytes.length == 0) {
-            Argon2SecureHasher a2sh = new Argon2SecureHasher();
-
             // Derive the reusable HMAC key from the nifi.sensitive.props.key to ensure deterministic output across nodes
             try {
+                NiFiPropertiesLoader.loadDefaultWithKeyFromBootstrap().getProperty(NiFiProperties.SENSITIVE_PROPS_ALGORITHM);
                 String npsk = NiFiPropertiesLoader.loadDefaultWithKeyFromBootstrap().getProperty(NiFiProperties.SENSITIVE_PROPS_KEY);
```

Comment:
Although it will require more adjustments, what do you think about refactoring to remove this approach to loading the NiFiProperties directly?  The SecureHasher reference could actually be eliminated entirely if the `sensitivePropertyKeyBytes` were computed based and passed as a constructor argument to `FingerprintFactory`.  At minimum, recommend refactoring the instantiation in `FlowFingerprintCheck` and `PopularVoteFlowElectionFactoryBean` to provide the sensitive properties key, but on review, it seems like the best approach is to refactor out the `SecureHasher` reference entirely and instead seed `FingerprintFactory` with the results of `SecureHasher.hashRaw(niFiSensitivePropertiesKey)`.